crates/viewer/re_renderer/shader/world_grid.wgsl

```diff
@@ -0,0 +1,154 @@
+    out.scaled_world_plane_position = plane_position / config.spacing;
+
+    return out;
+}
+
+// Like smoothstep, but linear.
+// Used for antialiasing: smoothstep works as well but is subtly worse.
+fn linearstep(edge0: f32, edge1: f32, x: f32) -> f32 {
+    return saturate((x - edge0) / (edge1 - edge0));
+}
+fn linearstep2(edge0: vec2f, edge1: vec2f, x: vec2f) -> vec2f {
+    return saturate((x - edge0) / (edge1 - edge0));
+}
+
+// Distance to a grid line in x and y ranging from 0 to 1.
+fn calc_distance_to_grid_line(scaled_world_plane_position: vec2f) -> vec2f {
+    return 1.0 - abs(fract(scaled_world_plane_position) * 2.0 - 1.0);
+}
+
+@fragment
+fn main_fs(in: VertexOutput) -> @location(0) vec4f {
+    // Most basics are very well explained by Ben Golus here: https://bgolus.medium.com/the-best-darn-grid-shader-yet-727f9278b9d8
+    // We're not actually implementing the "pristine grid shader" which is a grid with world space thickness,
+    // but rather the pixel space grid, which is a lot simpler, but happens to be also described very well in this article.
+
+    // Distance to a grid line in x and y ranging from 0 to 1.
+    let distance_to_grid_line = calc_distance_to_grid_line(in.scaled_world_plane_position);
+
+    // Figure out the how wide the lines are in this "draw space".
+    let plane_unit_pixel_derivative = fwidthFine(in.scaled_world_plane_position);
+    let line_anti_alias = plane_unit_pixel_derivative;
+    let width_in_pixels = 1.0;//config.thickness_ui * frame.pixels_from_point;
+    let width_in_grid_units = width_in_pixels * plane_unit_pixel_derivative;
+    var intensity_regular = linearstep2(width_in_grid_units + line_anti_alias, width_in_grid_units - line_anti_alias, distance_to_grid_line);
+
+    // Fade lines that get too close to each other.
+    // Once the number of pixels per line (== from one line to the next) is below a threshold fade them out.
+    //
+    // Note that `1/plane_unit_pixel_derivative` would give us more literal pixels per line,
+    // but we actually want to know how dense the lines get here so we use `1/width_in_grid_units` instead,
+    // such that a line density of 1.0 means roughly "100% lines" and 10.0 means "Every 10 pixels there is a lines".
+    // Empirically (== making the fade a hard cut and taking screenshot), this works out pretty accurately!
+    //
+    // Tried smoothstep here, but didn't feel right even with lots of range tweaking.
+    let line_density = 1.0 / max(width_in_grid_units.x, width_in_grid_units.y);
+    let grid_closeness_fade = linearstep(1.0, 10.0, line_density);
+    intensity_regular *= grid_closeness_fade;
+
+    // Every tenth line is a more intense, we call those "cardinal" lines.
+    // Experimented previously with more levels of cardinal lines, but it gets too busy:
+    // It seems that if we want to go down this path, we should ensure that there's only two levels of lines on screen at a time.
+    const CARDINAL_LINE_FACTOR: f32 = 10.0;
```

Comment:
For a reference: in `egui_plot` I have an infinite level of lines, but with only three levels viisble at any time, with a smooth fade to avoid discontinuities: https://emilk.github.io/egui_plot/

This means no matter how small or big your plots are, you always have thicker and thinner lines to guide you